lib/azure/armrest/model/base_model.rb

```diff
@@ -33,10 +33,21 @@ class BaseModel < Delegator
       #
       def initialize(json)
         @json = json
+        @resource_group = nil
         @ostruct = JSON.parse(json, object_class: OpenStruct)
         __setobj__(@ostruct)
       end
 
+      # Return the resource group for the current object.
+      def resource_group
+        @resource_group ||= id[/resourceGroups\/(.+?)\//i, 1] rescue nil
+      end
+
+      # Return a hash of tags associated with the resource.
+      def tags
```

Comment:
You can have tags, which are key/value pairs, on just about any Azure resource, either via the API or the portal. If there are no tags it will return an empty hash instead.

I've attached some sample tags to a few of my resources that you can see, such as the "miqazure-win2k8" VM, which shows "redhat => True".